lib-extra/src/main/java/com/diffplug/spotless/extra/ExtFormatterState.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra;
+
+import java.io.File;
+import java.io.Serializable;
+import java.lang.reflect.Method;
+import java.util.Objects;
+import java.util.Properties;
+
+import com.diffplug.spotless.FileSignature;
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterProperties;
+import com.diffplug.spotless.JarState;
+
+/**
+ * Creates optional formatter
+ *
+ * Serializes full state of the formatter to support Gradle incremental build.
+ */
+public final class ExtFormatterState implements Serializable {
+	private static final long serialVersionUID = 1L;
+
+	private static final String DEFAULT_FORMATTER_METHOD = "format";
+
+	private final JarState jarState;
+	private final FileSignature configFileSignature;
+	private final String className;
+	private String formatterMethod;
+
+	private ExtFormatterState(
+			final JarState jarState,
+			final String className,
+			final FileSignature configFileSignature) {
+		this.jarState = jarState;
+		this.configFileSignature = configFileSignature;
+		this.className = className;
+		this.formatterMethod = DEFAULT_FORMATTER_METHOD;
```

Comment:
It's not possible to change the value of the formatterMethod field.  Why is it an instance method?